contracts/BananaAllocator.sol

```diff
@@ -0,0 +1,235 @@
+    /// @param amount the amount of tokens to be withdrawn
+    function withdrawAllocation(
+        uint256 aid, 
+        uint256 amount
+    ) external onlyAllocationAdmin(aid, msg.sender) {
+        if(amount > 0 && amount <= allocationInfo[aid].tokensAvailable) {
+            allocationInfo[aid].tokensAvailable = allocationInfo[aid].tokensAvailable.sub(amount);
+            allocationToken.transfer(msg.sender, amount);
+
+            uint256 allocationTokenUsdValue = priceGetter.getTokenUsdPrice(address(allocationToken));
+            uint256 usdAmount = allocationTokenUsdValue.mul(amount).div(1e18);
+
+            emit AllocationWithdrawn(aid, amount, usdAmount, msg.sender);
+        }
+
+        if (lastSyncTimestamp + syncDelay < block.timestamp) {
+            syncAllocations();
+        }
+    }
+
+    /// @notice A function to withdraw from an allocation to another address
+    /// @param aid number referencing the applicable allocation
+    /// @param amount the amount of tokens to be withdrawn
+    /// @param to the address to withdraw tokens to
+    function withdrawAllocationTo(
+        uint256 aid, 
+        uint256 amount, 
+        address to
+    ) external onlyAllocationAdmin(aid, msg.sender) {
+        if(amount > 0 && amount <= allocationInfo[aid].tokensAvailable) {
+            allocationInfo[aid].tokensAvailable = allocationInfo[aid].tokensAvailable.sub(amount);
+            allocationToken.transfer(to, amount);
+
+            uint256 allocationTokenUsdValue = priceGetter.getTokenUsdPrice(address(allocationToken));
+            uint256 usdAmount = allocationTokenUsdValue.mul(amount).div(1e18);
+
+            emit AllocationWithdrawn(aid, amount, usdAmount, msg.sender);
+        }
+
+        if (lastSyncTimestamp + syncDelay < block.timestamp) {
+            syncAllocations();
+        }
+    }
+
+    /// @notice An internal function to sync all the allocation amounts
+    function syncAllocations() internal {
+        uint256 length = allocationInfo.length;
+        uint256 totalTokensOwed = 0;
+        uint256 tokensInContract = allocationToken.balanceOf(address(this));
+
+        for (uint256 aid = 0; aid < length; aid++) {
+            totalTokensOwed = totalTokensOwed.add(allocationInfo[aid].tokensAvailable);
+        }
+
+        uint256 tokensToAllocate = tokensInContract.sub(totalTokensOwed);
+        
+        for (uint256 i = 0; i < length; i++) {
+            allocationInfo[i].tokensAvailable += allocationInfo[i].allocationAmount.mul(tokensToAllocate).div(totalAllocations);
+        }
+
+        lastSyncTimestamp = block.timestamp;
+        
+        emit AllocationsSynced();
+    }
+
+    /// @notice A function to adjust the sync delay
+    /// @param _syncDelay number to adjust the sync delay to
+    ///     in milliseconds
+    function setSyncDelay(uint256 _syncDelay) external onlyOwner {
+        syncDelay = _syncDelay;
+
+        emit SyncDelaySet(_syncDelay);
```

Comment:
A good strategy for events which update a value is to call the event first and show the previous and current values as such: 

```
    /// event SyncDelaySet(uint256 previousSyncDelay, uint256 newSyncDelay);
    emit SyncDelaySet(syncDelay, _syncDelay);
    syncDelay = _syncDelay;
```

Reply:
Easy fix thanks ser